### PR TITLE
man: document special considerations when mixing templated service un…

### LINKED ISSUE #2189: 227 regression: Instantiated units always get stopped during shutdown
In Debian we have a helper unit `ifup@.service` which gets triggered by udev when a network device appears, and then it starts the corresponding network interface if that's configured by ifupdown (i. e. not networkd). The idea is that we never want this to get stopped during shutdown, as this breaks e. g. root on NFS or on iSCSI. It's also not necessary as there already is a `networking.service` which shuts down the network interfaces (unless they are still in use in the cases mentioned).

227 introduced a regression that instantiated `foo@.service` units now always get stopped during shutdown even if they have `DefaultDependencies=no` and thus no implied `Conflicts=shutdown.service`.

This can be demonstrated in nspawn with a `/etc/systemd/system/letmelive@.service`:

```
[Unit]
Description=@@@@@@@@@@@@@@@@@@@ Let me live %i
DefaultDependencies=no

[Service]
RemainAfterExit=yes
StandardOutput=journal+console
ExecStart=/bin/true
ExecStop=/bin/echo "!!!!!!!!!!! I GOT KILLED !!!!!!!!!!!!!"
```

(The ! and @ make it much easier to recognize in the scrolled output) and `systemctl start letmelive@1.service`. https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1492546/+attachment/4536638/+files/systemd-letmelive.sh provides a fully automatic test case which takes a container root directory as the (only) argument.

Using this, I bisected  this to https://github.com/systemd/systemd/commit/8c8da0e0cb4. After that, `letmelive@1.service` has

```
Requires=system-letmelive.slice
After=systemd-journald.socket system-letmelive.slice
```

and I suppose `system-letmelive.slice` is always torn down during shutdown.

This new dependency cannot be reset using `Requires=` in the unit file, as this gets added in code.

The commit log of 8c8da0e0cb4 is not very clear to me, but I guess it fixed some real issue.

Is there any way to write instantiated units so that they don't get stopped during shutdown? (Note that this does work for "normal" units without templates, i. e. no `@`).

Or is this use case simply not supported, and we need to drop the `ExecStop=` in our `ifup@.service`? This would mean that unplugging the device  wouldn't stop and properly deconfigure an interface. Or we need to make `ExecStop=` more complicated to not actually deconfigure when the system is shutting down; both are a bit ugly.

### PR DESCRIPTION
**Pull request by poettering**
…its and DefaultDependencies=no

Fixes #2189.

### CODE REVIEW

File: man/systemd.service.xml [hunk 1]
```diff
@@ -113,6 +113,18 @@
     involved with early boot or late system shutdown should disable
     this option.</para>
 
+    <para>Note that instanced service units (i.e. service units with an <literal>@</literal> in their name) are by
+    default assigned a per-template slice unit (see
```
Comment (martinpitt) [inline man/systemd.service.xml]: The "Note that" is mostly just filling, and could go.

I believe "are assigned [...] by default" is better grammar.

File: man/systemd.service.xml [hunk 2]
```diff
@@ -113,6 +113,18 @@
...
+    <citerefentry><refentrytitle>systemd.slice</refentrytitle><manvolnum>5</manvolnum></citerefentry>), named after the
+    template unit, containing all instances of the specific template. This slice is normally stopped at shutdown. Also
+    note that service units require their slice units, and hence if a slice unit is stopped all service units contained
+    in it are stopped, too. This means if you write a template service unit that shall stay around during late
+    shutdown, and is using <varname>DefaultDependencies=no</varname> to implement this, make sure to either define your
+    own per-template slice unit file that also sets <varname>DefaultDependencies=no</varname>, or to explicitly assign
+    your templated service unit to a different slice unit that already sets this, for example by using
+    <varname>Slice=system.slice</varname>. Also see
+    <citerefentry><refentrytitle>systemd.resource-control</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para>
```
Comment (martinpitt) [inline man/systemd.service.xml:124]: The explanation/solutions look correct (I tested both), but it's IMHO a bit too technical to understand. Maybe

This slice is normally stopped at shutdown, _together with all template instances. If that is not desired, set  <varname>DefaultDependencies=no</varname> in the template unit, and either_ define your own per-template slice unit file that also sets <varname>DefaultDependencies=no</varname>, _or set <varname>Slice=system.slice</varname> (or another suitable slice) in the template unit._ Also see...

**Comment by martinpitt** [comment]
Looks technically correct, I just have two grammar nitpicks and a proposal for trimming the explanation a bit.

File: man/systemd.service.xml (hunk 1 above)
Comment (poettering) [inline man/systemd.service.xml]: ok, will fix.

**Comment by martinpitt** [comment]
LGTM now, danke!

**Pull request merged**